src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = "::";
+    private static final String COMA = ",";
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final int ACTION_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readFromFile(fromFileName);
+        writeToFile(toFileName, generateReport(data));
+    }
+
+    private String generateReport(String[] data) {
+        String[] tmpArrStr;
+        int buy = 0;
+        int supply = 0;
+        for (String s : data) {
+            tmpArrStr = s.split(",");
+            switch (tmpArrStr[ACTION_INDEX]) {
+                case BUY:
+                    buy += Integer.parseInt(tmpArrStr[AMOUNT_INDEX]);
+                    break;
+                case SUPPLY:
+                    supply += Integer.parseInt(tmpArrStr[AMOUNT_INDEX]);
+                    break;
+                default:
+                    break;
+            }
+        }
+        return SUPPLY + COMA + supply + System.lineSeparator()
+                + BUY + COMA + buy + System.lineSeparator()
+                + RESULT + COMA + (supply - buy);
```

Comment:
lets use StringBuilder instead of concatenation for better performance

Reply:
Ide proposed. First i did it trow StringBuilder